modules/juce_audio_devices/native/juce_ios_Audio.cpp

```diff
@@ -1012,10 +1015,13 @@ struct iOSAudioIODevice::Pimpl      : public AudioPlayHead,
         desc.componentManufacturer = kAudioUnitManufacturer_Apple;
         desc.componentFlags = 0;
         desc.componentFlagsMask = 0;
-                
+               
+        setAudioPreprocessingEnabled(false);

```

Comment:
Hmm this seems like it should be true, no? Maybe it also belongs in an else block below instead to avoid switching the session category too much?

Reply:
This is just to use AVAudioSessionModeMeasurement with headphones. It doesn't lower the volume like when we use the speaker, so the gain compensation is 1.0 for this case. It might be good to use this mode to lower the latency

Reply:
I can leave this change out for now and use AVAudioSessionModeMeasurement just in no-headphones case

Reply:
hmm ok, it might be good to add some debug logs here for good measure just to know the exact state we're choosing here. Also, you can probably remove line 1026 then, right?

Reply:
> Should this also be wrapped in an `if (!AudioIODeviceType::useDeviceVoiceProcessing)`?

I thought we wanted to use `AVAudioSessionModeMeasurement` for both headphones and non headphones states. I'll add the debug logs

Reply:
useDeviceVoiceProcessing isn't relevant to headphone state actually. It's just about whether vpio is supopsed to be used when appropriate

Reply:
True, I was looking at `if (isUsingBuiltInSpeaker() && !AudioIODeviceType::useDeviceVoiceProcessing)`. I'll add the condition